您好，可以写一份 store.d.ts文件吗？

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build:test"
npm ERR! node v6.10.0
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! storejs@1.0.16 build:test: `browserify test/store.js > test/store.test.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the storejs@1.0.16 build:test script 'browserify test/store.js > test/store.test.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the storejs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     browserify test/store.js > test/store.test.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs storejs
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls storejs
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     E:\zqVue\store.js\npm-debug.log

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build" "--scripts-prepend-node-path=auto"
npm ERR! node v6.10.0
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! storejs@1.0.16 build: `npm run build:min && npm run build:dist && npm run build:test`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the storejs@1.0.16 build script 'npm run build:min && npm run build:dist && npm run build:test'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the storejs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build:min && npm run build:dist && npm run build:test
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs storejs
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls storejs
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     E:\zqVue\store.js\npm-debug.log

--- a/jest.config.js
+++ b/jest.config.js
@@ -1,6 +1,7 @@
 // jest.config.js
 module.exports = {
   verbose: true,
+  testPathIgnorePatterns: ['/node_modules/', '/types/'],
   collectCoverageFrom: [
     "**/dist/store.common.js"
   ]

--- a/package.json
+++ b/package.json
@@ -10,11 +10,12 @@
     "type": "git",
     "url": "https://github.com/jaywcjlove/store.js.git"
   },
+  "types": "types/index.d.ts",
   "scripts": {
     "prepare": "npm run build",
     "build": "node scripts/build.js",
     "watch": "node scripts/watch.js",
-    "test": "jest --coverage",
+    "test": "jest --coverage && tsc -p ./types",
     "test:watch": "jest --coverage --watch"
   },
   "keywords": [
@@ -25,7 +26,8 @@
     "localstorage"
   ],
   "files": [
-    "dist/*.js"
+    "dist/*.js",
+    "types/index.d.ts"
   ],
   "author": "kenny wang <wowohoo@qq.com>",
   "license": "MIT",
@@ -49,12 +51,14 @@
     "@babel/preset-env": "^7.7.7",
     "@rollup/plugin-commonjs": "^11.0.0",
     "@rollup/plugin-node-resolve": "^6.0.0",
+    "@type-challenges/utils": "^0.1.1",
     "bannerjs": "^1.0.7",
     "colors-cli": "^1.0.26",
     "jest": "^24.9.0",
     "rollup": "^1.27.14",
     "rollup-plugin-babel": "^4.3.3",
     "rollup-plugin-terser": "^5.1.3",
+    "typescript": "^4.1.3",
     "zlib": "^1.0.5"
   },
   "dependencies": {}

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -1,0 +1,36 @@
+declare function store(): Record<string, any>
+declare function store(key: `?${string}`): boolean
+declare function store(key: string): any
+declare function store(key: string, value: string): Store
+declare function store(entries: Record<string, any>): Store
+declare function store<S extends string>(key: S, callback: (key: S, value: any) => any): Store
+declare function store<S extends readonly string[]>(keys: S, callback: (key: S[number], value: any) => any): Store
+
+/** @deprecated Don't pass falsy value in this way. Use `set` instead. */
+declare function store(key: string, value: Falsy): Store
+type Falsy = null | undefined | 0 | '' | false
+
+declare class Store {
+  get(): Record<string, any>
+  get(key: `?${string}`): boolean
+  get(key: string): any
+  get<S extends readonly string[]>(...keys: S): Record<S[number], any>
+
+  set(key: string, value: any): Store
+  set(entries: Record<string, any>): Store
+
+  has(key: string): boolean
+
+  remove(key: string): any
+
+  keys(): string[]
+
+  forEach(callback: (key: string, value: any) => void): Store
+
+  search(keyword: string): Record<string, any>
+
+  clear(): Store
+}
+
+declare let s: typeof store & Store
+export default s

--- a/types/test.ts
+++ b/types/test.ts
@@ -1,0 +1,41 @@
+import store from '.'
+import type { Equal, Expect } from '@type-challenges/utils'
+
+let s1: any = store().test
+let v1: string = store('k1')
+let exists1 = store('?k')
+type exists1 = Expect<Equal<typeof exists1, boolean>>
+
+store('k1', 'v1').set('k2', 'v2')
+store({ k1: 'v1', k2: 'v2' }).set('k3', 'v3')
+
+store('key', (key) => {
+  type k = Expect<Equal<'key', typeof key>>
+})
+store(['k1', 'k2'] as const, (key) => {
+  type k = Expect<Equal<'k1' | 'k2', typeof key>>
+})
+
+let v2: any = store.get().k2
+let v3: string = store.get('k3')
+let exists2 = store.get('?k')
+type exists2 = Expect<Equal<typeof exists2, boolean>>
+let entries1 = store.get('a', 'b', 'c')
+type keys1 = Expect<Equal<keyof typeof entries1, 'a' | 'b' | 'c'>>
+
+store.set({ k1: 'v1', k2: 'v2' }).set('k3', 'v3')
+
+let exists3 = store.has('k')
+type exists3 = Expect<Equal<typeof exists3, boolean>>
+
+let v4: any = store.remove('k4')
+
+let keys = store.keys()
+type keys = Expect<Equal<typeof keys, string[]>>
+
+store.clear()
+  .forEach((key, value: any) => {
+    type key = Expect<Equal<typeof key, string>>
+  })
+
+let k5: any = store.search('k').k5

--- a/types/tsconfig.json
+++ b/types/tsconfig.json
@@ -1,0 +1,8 @@
+{
+  "compilerOptions": {
+    "strict": true,
+    "module": "es2015",
+    "moduleResolution": "node",
+    "noEmit": true
+  },
+}

